gammapy/estimators/parameter.py

```diff
@@ -345,5 +377,107 @@ def run(self, datasets, parameter):
             if "scan" in self.selection_optional:
                 result.update(self.estimate_scan(datasets, parameter))
 
+            if "sensitivity" in self.selection_optional:
+                result.update(self.estimate_sensitivity(datasets, parameter))
+
         result.update(self.estimate_counts(datasets))
         return result
+
+
+class ParameterSensitivityEstimator:
+    """Estimate the sensitivity to a given parameter
+
+    Computes the TS distribution in the non-null hypothesis using the
+    log likelihood of the Asimov dataset (i.e. a dataset with counts = npred)
+    and the non central chi2 distribution.
+    Once the TS distribution under the testing hypothesis is known,
+    one can compute the required parameter value
+    to have 50% of measurements above a given significance threshold.
+
+
+    Parameters
+    ----------
+    parameter : `~gammapy.modeling.Parameter`
+       Parameter to test
+    null_value : float or `~gammapy.modeling.Parameter`
+        Value of the parameter for the null hypothesis.
+    n_sigma : int, default=5
+        Number of required significance level.
+    rtol : float
+        Relative precision of the estimate. Used as a stopping criterion.
+        Default is 0.01.
+    max_niter : int
+        Maximal number of iterations used by the root finding algorithm.
+        Default is 100.
+
+    References
+    ----------
+        Cowan et al. (2011), European Physical Journal C, 71, 1554.
+        doi:10.1140/epjc/s10052-011-1554-0.
+
+    """
+
+    tag = "ParameterSensitivityEstimator"
+
+    def __init__(
+        self,
+        parameter,
+        null_value,
+        n_sigma=5,
+        n_free_parameters=None,
+        rtol=0.01,
+        max_niter=100,
+    ):
+        self.test = TestStatisticNested(
+            [parameter], [null_value], n_free_parameters=n_free_parameters
+        )
```

Comment:
A general design question here. You might have moved most the content of the ParameterSensitivityEstimator into the `TestStatisticNested`. Could it make sense?

Reply:
I think that the ParameterSensitivityEstimator is higher level than the TestStatisticNested, and how rtol and max_niter option should be passed if moved on TestStatisticNested ? as a kwarg_sensitivity (seems more complicated to me)